MapboxNavigation/StyleManager.swift

```diff
@@ -116,6 +116,7 @@ open class StyleManager: NSObject {
         guard let location = delegate?.locationFor(styleManager: self) else {
             // We can't calculate sunset or sunrise w/o a location so just apply the first style
             if let style = styles.first {
+                currentStyleType = style.styleType
```

Comment:
@bsudekum you also check: `currentStyleType != styleType` before updating the `currentStyleType` and invoking the `style .apply()` and `delegate?.styleManager?(self, didApply: style)`.